extensions/amp-date-picker/0.1/amp-date-picker.js

```diff
@@ -0,0 +1,703 @@
+   */
+  hasDayTemplate_() {
+    return this.templates_.hasTemplate(this.element, '[date-template]');
+  }
+
+  /**
+   * @param {!Array<!DateTemplateMapDef>} templates
+   * @param {!moment} date
+   * @return {?Element}
+   */
+  getTemplate_(templates, date) {
+    for (let i = 0; i < templates.length; i++) {
+      if (templates[i].dates.contains(date)) {
+        return templates[i].template;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Get the template tag corresponding to a given date.
+   * @param {!moment} date
+   * @return {?Element}
+   */
+  getDayTemplate_(date) {
+    return (
+      this.getTemplate_(this.srcTemplates_, date) ||
+      this.getTemplate_(this.elementTemplates_, date) ||
+      this.srcDefaultTemplate_ ||
+      scopedQuerySelector(this.element, '[date-template][default]')
+    );
+  }
+
+  /**
+   * Render the template that corresponds to the date with its data.
+   * @param {!moment} day
+   * @param {!JsonObject} data
+   */
+  renderDayTemplate_(day, data) {
+    const template = this.getDayTemplate_(day);
+    return this.renderTemplate_(template, data, day.format('D'));
+  }
+
+  /**
+   * Render the info section of the calendar view.
+   */
+  renderInfo() {
+    if (!this.infoTemplatePromise_) {
+      this.infoTemplatePromise_ = this.renderInfoTemplate_();
+    }
+    this.scanForBindings_([this.container_]);
+    return this.renderPromiseIntoReact_(this.infoTemplatePromise_);
+  }
+
+  /**
+   * Returns `true` if an info template exists.
+   * @return {boolean}
+   */
+  hasInfoTemplate_() {
+    return this.templates_.hasTemplate(this.element, '[info-template]');
+  }
+
+  /** @return {!Promise<string>} */
+  renderInfoTemplate_() {
+    const template = scopedQuerySelector(this.element, '[info-template]');
+    return this.renderTemplate_(template);
+  }
+
+  /**
+   * @param {!Array<!Element>} elements
+   * @return {!Promise<!Array<!Element>>}
+   * @private
+   */
+  scanForBindings_(elements) {
+    const forwardElements = () => elements;
+    return Services.bindForDocOrNull(this.element).then(bind => {
+      if (bind) {
+        return bind.rescanAndEvaluate(elements);
+      }
+    // Forward elements to chained promise on success or failure.
+    }).then(forwardElements, forwardElements);
+  }
+
+  /**
+   * Render the given template with the given data. If the template does not
+   * exist, use a fallback string.
+   * The fallback string will be rendered directly into the DOM so it must
+   * not contain unsanitized user-supplied values.
+   * @param {?Element} template
+   * @param {!JsonObject=} opt_data
+   * @param {string=} opt_fallback
+   * @return {!Promise<string>}
+   */
+  renderTemplate_(template, opt_data, opt_fallback) {
+    if (template) {
+      const data = opt_data || /** @type {!JsonObject} */ ({});
+      return this.templates_.renderTemplate(template, data)
+          .then(rendered => {
+            rendered.setAttribute('i-amphtml-rendered', '');
```

Comment:
you probably don't need to set this unless you need to know which once have been rendered to clear them,